system/Validation/FormatRules.php

```diff
@@ -128,10 +128,12 @@ public function hex(?string $str = null): bool
 
     /**
      * Integer
+     *
+     * @param int|string|null $str
      */
-    public function integer(?string $str = null): bool
+    public function integer($str = null): bool
     {
-        return (bool) preg_match('/\A[\-+]?\d+\z/', $str ?? '');
+        return is_int($str);
```

Comment:
All data sent via HTTP is represented as strings and arrays. That is, an integer as a string will not pass validation.